arcgis-ios-sdk-samples/Route and directions/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -0,0 +1,151 @@
+        mockNMEADataSource.start()
+        mapView.locationDisplay.start()
+        // Recenter the map and set pan mode.
+        recenter()
+    }
+    
+    @IBAction func recenter() {
+        mapView.locationDisplay.autoPanMode = .recenter
+        recenterBarButtonItem.isEnabled = false
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in
+            DispatchQueue.main.async { [weak self] in
+                self?.recenterBarButtonItem.isEnabled = true
+            }
+            self?.mapView.locationDisplay.autoPanModeChangedHandler = nil
+        }
+    }
+    
+    @IBAction func reset() {
+        // Reset buttons states.
+        startBarButtonItem.isEnabled = true
+        resetBarButtonItem.isEnabled = false
+        // Reset the status text.
+        statusLabel.text = "Satellites info will be shown here."
+        // Reset and stop the location display.
+        mapView.locationDisplay.autoPanModeChangedHandler = nil
+        mapView.locationDisplay.autoPanMode = .off
+        mapView.locationDisplay.stop()
+        // Pause the mock data generation.
+        mockNMEADataSource.stop()
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "DisplayDeviceLocationWithNMEADataSourcesViewController",
+            "SimulatedNMEADataSource"
+        ]
+        // Load NMEA location data source.
+        startBarButtonItem.isEnabled = true
+        nmeaLocationDataSource.locationChangeHandlerDelegate = self
+        mockNMEADataSource.delegate = self
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        reset()
+    }
+}
+
+// MARK: SimulatedNMEADataSourceDelegate
+
+extension DisplayDeviceLocationWithNMEADataSourcesViewController: SimulatedNMEADataSourceDelegate {
+    func dataSource(didUpdate nmeaData: Data) {
+        // Push mock data into the data source.
+        // Note: You can also get real-time NMEA sentences from a GPS dongle.
+        nmeaLocationDataSource.push(nmeaData)
+    }
+}
+
+// MARK: AGSNMEALocationDataSourceDelegate
+
+extension DisplayDeviceLocationWithNMEADataSourcesViewController: AGSNMEALocationDataSourceDelegate {
+    func nmeaLocationDataSource(_ NMEALocationDataSource: AGSNMEALocationDataSource, satellitesDidChange satellites: [AGSNMEASatelliteInfo]) {
+        // Update the satellites info status text.
+        let satelliteSystemsText = Set(satellites.map { $0.system }).map { $0.label }.joined(separator: ", ")
+        let idText = satellites.map { String($0.satelliteID) }.joined(separator: ", ")
+        let statusText =
+            """
+            \(satellites.count) satellites in view
+            System(s): \(satelliteSystemsText)
+            IDs: \(idText)
```

Comment:
Can there be only one ID?

```suggestion
            ID(s): \(idText)
```

Reply:
Not sure about this. The NMEA [spec](http://aprs.gids.nl/nmea/#gsv) doesn't say the range of satellite count in the GSV messages. But my intuition says we'll need [at least 3 satellites](https://www.quora.com/What-is-the-minimum-number-of-satellites-a-GPS-system-requires-in-order-to-function) to locate a point on the earth surface... 🤔 

The mock data always shows 12.

Reply:
That makes sense.